core/src/tasks/helpers.rs

```diff
@@ -0,0 +1,27 @@
+//! Helpers for common functionality across tasks
+
+use crate::{
+    starknet_client::{client::StarknetClient, error::StarknetClientError},
+    state::wallet::{Wallet, WalletAuthenticationPath},
+};
+
+/// Find the merkle authentication path of a wallet
+pub(super) async fn find_merkle_path(
```

Comment:
do relayers maintain a synced local version of the merkle tree? to avoid only being able to query the contract for inclusion proofs

Reply:
Not currently; we may maintain a sparse representation in the future as an optimization, but for now they just recover the paths they need from contract state on demand.